PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -276,7 +276,18 @@ public function select($sec, $usec)
         $write = null;
         $except = null;
 
-        return stream_select($read, $write, $except, $sec, $usec);
+        $result = @stream_select($read, $write, $except, $sec, $usec);
```

Comment:
The problem: stream_select generates a warning when app is interrupted by a signal.
So if you have a custom error handler that treats warnings as fatal then app crashes.